### PR TITLE
Build  update typescript to 3.4.x and rx js to 6.5.x

### LINKED ISSUE #1788: Build: Update TypeScript to 3.4.x and RxJS to 6.5.x
Update the versions for the dependencies so we'll be using the latest that the Angular framework uses upon release.

- Update necessary `package.json` files for the repo and examples
- Update Bazel [dependency versions](https://github.com/ngrx/platform/blob/master/tools/defaults.bzl#L10)
- Update the [migration guide](https://github.com/ngrx/platform/blob/master/projects/ngrx.io/content/guide/migration/v8.md) for v8 with the new versions
- Fix any necessary changes for TS 3.4.x

### PR DESCRIPTION
**Pull request by santoshyadav198613**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #1788

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by santoshyadav198613** [comment]
WIP, please do not review

**Comment by santoshyadav198613** [comment]
Hi @brandonroberts ,
I have updated the package, stuck with one issue, external schematics is not generating the workspace, due to which test cases are getting failed.

Can you please guide i even tried to refer 
https://github.com/angular/angular/blob/f8096d499324cf0961f092944bbaedd05364eea1/packages/elements/schematics/ng-add/index_spec.ts

**Comment by timdeschryver** [comment]
I got this solved but it seems like I can't push to this branch 🤔

**Comment by santoshyadav198613** [comment]
> I got this solved but it seems like I can't push to this branch

Thanks Tim, it worked now.

**Comment by santoshyadav198613** [comment]
> I got this solved but it seems like I can't push to this branch

Hi @timdeschryver ,
This time it failed for bazel test step,no idea what wrong i am doing here, sorry for bothering you. :(

**Comment by santoshyadav198613** [comment]
> > I got this solved but it seems like I can't push to this branch
> 
> Hi @timdeschryver ,
> This time it failed for bazel test step,no idea what wrong i am doing here, sorry for bothering you. :(

I think it has something to do with bazel, i am referring to https://github.com/angular/angular/blob/master/WORKSPACE
Hope it's get resolved.

**Comment by timdeschryver** [comment]
We probably have to update some bazel files 😅
I'll try to take a look at it tomorrow.

**Comment by santoshyadav198613** [comment]
> We probably have to update some bazel files 
> I'll try to take a look at it tomorrow.

Thanks, meanwhile am trying by referring angular repo.

File: modules/schematics-core/testing/create-workspace.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import {
-  UnitTestTree,
   SchematicTestRunner,
+  UnitTestTree,
 } from '@angular-devkit/schematics/testing';
+import { concatMap } from 'rxjs/internal/operators/concatMap';
```
Comment (timdeschryver) [inline modules/schematics-core/testing/create-workspace.ts]:
```suggestion
import { concatMap } from 'rxjs/operators';
```

**Comment by santoshyadav198613** [comment]
Not sure if this is the issue:
https://www.npmjs.com/package/ecstatic
this guy removed all the previous packages, but in that case how the build worked so far in the last 5 days.

File: projects/ngrx.io/package.json [hunk 2]
```diff
@@ -123,7 +123,7 @@
     "hast-util-is-element": "^1.0.0",
     "hast-util-to-string": "^1.0.0",
     "html": "^1.0.0",
-    "http-server": "^0.9.0",
+    "http-server": "^0.11.1",
```
Comment (timdeschryver) [inline projects/ngrx.io/package.json]: We had to update these because of a dependency removal - for more info see https://github.com/indexzero/http-server/issues/521

**Comment by santoshyadav198613** [comment]
Open for review now.

**Comment by santoshyadav198613** [comment]
> I noticed that the changes from #1805 are also in this PR, is this OK?
> The rest of this PR LGTM

Done @timdeschryver that was there by mistake, rectified now.

**Comment by brandonroberts** [comment]
I wonder why these tests needed to be converted to be async in the first place. Instead of using subscribes, add `toPromise()` to the `runSchematicAsync` and use async/await. Reference https://github.com/angular/angular-cli/blob/master/packages/schematics/angular/component/index_spec.ts

**Comment by brandonroberts** [comment]
This line needs to be updated to ^6.5.0 also https://github.com/ngrx/platform/blob/master/tools/defaults.bzl

**Comment by santoshyadav198613** [comment]
> This line needs to be updated to `^6.5.0` also https://github.com/ngrx/platform/blob/master/tools/defaults.bzl

https://github.com/ngrx/platform/pull/1795/files#diff-653388f9d55f0ae80bba217809819c24R10
it is updated.

**Comment by santoshyadav198613** [comment]
> I wonder why these tests needed to be converted to be async in the first place. Instead of using subscribes, add `toPromise()` to the `runSchematicAsync` and use async/await. Reference https://github.com/angular/angular-cli/blob/master/packages/schematics/angular/component/index_spec.ts

Hi @brandonroberts ,

After updating to the beta version, we started getting errors to convert external schematics to async.

**Comment by timdeschryver** [comment]
🤯 ugh, so much cleaner. My bad ... 😅
I found https://github.com/angular/angular/blob/f8096d499324cf0961f092944bbaedd05364eea1/packages/elements/schematics/ng-add/index_spec.ts#L44 but didn't notice it was old.

**Comment by santoshyadav198613** [comment]
> 🤯 ugh, so much cleaner. My bad ... 
> I found https://github.com/angular/angular/blob/f8096d499324cf0961f092944bbaedd05364eea1/packages/elements/schematics/ng-add/index_spec.ts#L44 but didn't notice it was old.

Yes let me do this, so after this code we don't need to subscribe any more much cleaner code

**Comment by santoshyadav198613** [comment]
Hi @timdeschryver and @brandonroberts ,

I have changed the code, 6 test cases are failing will take a look into it tomorrow.

**Comment by santoshyadav198613** [comment]
Update failing test case for reducer only.

File: modules/schematics-core/testing/create-workspace.ts [hunk 3]
```diff
@@ -20,13 +22,6 @@ export const defaultAppOptions = {
   skipTests: false,
 };
 
-const defaultModuleOptions = {
```
Comment (brandonroberts) [inline modules/schematics-core/testing/create-workspace.ts]: Leave this code as its an unrelated change

File: modules/schematics/src/cli.spec.ts [hunk 4]
```diff
@@ -23,15 +23,17 @@ describe('CLI Schematic', () => {
...
+    appTree = await createWorkspace(schematicRunner, appTree);
   });
 
   it('should create a class by the angular/cli', () => {
     const options = { ...defaultOptions };
-    const tree = schematicRunner.runSchematic('class', options, appTree);
-    const content = tree.readContent(`${projectPath}/src/app/foo.ts`);
-
-    expect(content).toMatch(/export class Foo/);
+    schematicRunner
```
Comment (brandonroberts) [inline modules/schematics/src/cli.spec.ts]: No subscribe, use async/await

File: modules/schematics/src/container/index.spec.ts [hunk 5]
```diff
@@ -32,79 +32,102 @@ describe('Container Schematic', () => {
...
   });
 
   it('should respect the state option if not provided', () => {
     const options = { ...defaultOptions, state: undefined };
-    const tree = schematicRunner.runSchematic('container', options, appTree);
-    const content = tree.readContent(
-      `${projectPath}/src/app/foo/foo.component.ts`
-    );
-    expect(content).not.toMatch(/import \* as fromStore/);
+    schematicRunner
```
Comment (brandonroberts) [inline modules/schematics/src/container/index.spec.ts]: No subscribe, use async/await

File: modules/schematics/src/container/index.spec.ts [hunk 6]
```diff
@@ -32,79 +32,102 @@ describe('Container Schematic', () => {
...
     const options = { ...defaultOptions, state: 'reducers/foo.ts' };
     appTree.create(`${projectPath}/src/app/reducers/foo.ts`, '');
-    const tree = schematicRunner.runSchematic('container', options, appTree);
-    const content = tree.readContent(
-      `${projectPath}/src/app/foo/foo.component.ts`
-    );
-    expect(content).toMatch(/import \* as fromStore from '..\/reducers\/foo';/);
+    schematicRunner
+      .runSchematicAsync('container', options, appTree)
+      .subscribe(tree => {
```
Comment (brandonroberts) [inline modules/schematics/src/container/index.spec.ts]: No subscribe, use async/await

File: modules/schematics/src/container/index.spec.ts [hunk 7]
```diff
@@ -32,79 +32,102 @@ describe('Container Schematic', () => {
...
     const options = { ...defaultOptions, state: 'reducers/index.ts' };
     appTree.create(`${projectPath}/src/app/reducers/index.ts`, '');
-    const tree = schematicRunner.runSchematic('container', options, appTree);
-    const content = tree.readContent(
-      `${projectPath}/src/app/foo/foo.component.ts`
-    );
-    expect(content).toMatch(/import \* as fromStore from '..\/reducers';/);
+    schematicRunner
+      .runSchematicAsync('container', options, appTree)
+      .subscribe(tree => {
```
Comment (brandonroberts) [inline modules/schematics/src/container/index.spec.ts]: No subscribe, use async/await

File: modules/schematics/src/container/index.spec.ts [hunk 8]
```diff
@@ -32,79 +32,102 @@ describe('Container Schematic', () => {
...
 
   it('should import Store into the component', () => {
     const options = { ...defaultOptions, state: 'reducers' };
     appTree.create(`${projectPath}/src/app/reducers`, '');
-    const tree = schematicRunner.runSchematic('container', options, appTree);
-    const content = tree.readContent(
-      `${projectPath}/src/app/foo/foo.component.ts`
-    );
-    expect(content).toMatch(/import\ {\ Store\ }\ from\ '@ngrx\/store';/);
+    schematicRunner
```
Comment (brandonroberts) [inline modules/schematics/src/container/index.spec.ts]: No subscribe, use async/await

File: modules/schematics/src/container/index.spec.ts [hunk 9]
```diff
@@ -32,79 +32,102 @@ describe('Container Schematic', () => {
...
-    const tree = schematicRunner.runSchematic('container', options, appTree);
-    const content = tree.readContent(
-      `${projectPath}/src/app/foo/foo.component.ts`
-    );
-    expect(content).toMatch(
-      /constructor\(private store\: Store\<fromStore\.State\>\) { }\n\n/
-    );
+    schematicRunner
+      .runSchematicAsync('container', options, appTree)
+      .subscribe(tree => {
```
Comment (brandonroberts) [inline modules/schematics/src/container/index.spec.ts]: No subscribe, use async/await

File: modules/schematics/src/container/index.spec.ts [hunk 10]
```diff
@@ -32,79 +32,102 @@ describe('Container Schematic', () => {
...
-    const tree = schematicRunner.runSchematic('container', options, appTree);
-    const content = tree.readContent(
-      `${projectPath}/src/app/foo/foo.component.spec.ts`
-    );
-    expect(content).toMatch(
-      /import { Store, StoreModule } from '@ngrx\/store';/
-    );
+    schematicRunner
+      .runSchematicAsync('container', options, appTree)
+      .subscribe(tree => {
```
Comment (brandonroberts) [inline modules/schematics/src/container/index.spec.ts]: No subscribe, use async/await

File: modules/schematics/src/reducer/index.spec.ts [hunk 11]
```diff
@@ -27,8 +27,8 @@ describe('Reducer Schematic', () => {
 
   let appTree: UnitTestTree;
 
-  beforeEach(() => {
-    appTree = createReducers(createWorkspace(schematicRunner, appTree));
+  beforeEach(async () => {
+    appTree = await createWorkspace(schematicRunner, appTree);
```
Comment (brandonroberts) [inline modules/schematics/src/reducer/index.spec.ts]: Why was `createReducers` removed?

File: tools/defaults.bzl [hunk 12]
```diff
@@ -3,11 +3,11 @@
 load("@build_bazel_rules_nodejs//:defs.bzl", _npm_package = "npm_package")
 load("@npm_angular_bazel//:index.bzl", _ng_module = "ng_module", _ng_package = "ng_package")
 load("@npm_bazel_jasmine//:index.bzl", _jasmine_node_test = "jasmine_node_test")
-load("@npm_bazel_typescript//:defs.bzl", _ts_library = "ts_library")
+load("@npm_bazel_typescript//:index.bzl", _ts_library = "ts_library")
 
 DEFAULT_TSCONFIG = "//:tsconfig.json"
 NG_VERSION = "^8.0.0-beta"
-RXJS_VERSION = "^6.4.0"
+RXJS_VERSION = "^6.5.1"
```
Comment (brandonroberts) [inline tools/defaults.bzl]:
```suggestion
RXJS_VERSION = "^6.5.0"
```

File: modules/schematics/src/reducer/index.spec.ts (hunk 11 above)
Comment (santoshyadav198613) [reply modules/schematics/src/reducer/index.spec.ts]: It was by mistake and this is why my test cases were failing, thanks for pointing it out.

**Comment by santoshyadav198613** [comment]
Hi @brandonroberts ,
Done with the changes. You can do the review now, all the test cases working now.

File: tools/defaults.bzl (hunk 12 above)
Comment (alex-okrushko) [reply tools/defaults.bzl]: why not `6.5.1`?

File: package.json [hunk 13]
```diff
@@ -76,35 +76,35 @@
...
-    "@angular/animations": "^8.0.0-beta.10",
-    "@angular/bazel": "^8.0.0-beta.10",
-    "@angular/cdk": "^7.3.0",
-    "@angular/cli": "^8.0.0-beta.10",
-    "@angular/common": "^8.0.0-beta.10",
-    "@angular/compiler": "^8.0.0-beta.10",
-    "@angular/compiler-cli": "^8.0.0-beta.10",
-    "@angular/core": "^8.0.0-beta.10",
-    "@angular/forms": "^8.0.0-beta.10",
+    "@angular/animations": "^8.0.0-beta.13",
```
[the remark this answers is not in the document]
Comment (santoshyadav198613) [reply package.json]: Done @brandonroberts I have updated the version, also in rc version, @angular/http dependency has been removed and the package is not available so removed the same.

**Comment by brandonroberts** [comment]
LGTM. Once Tim approves, we'll get it merged. Thanks for working through the changes!

**Comment by alex-okrushko** [comment]
@santoshyadav198613 Fantastic work! Thanks!

**Comment by santoshyadav198613** [comment]
> @santoshyadav198613 Fantastic work! Thanks!

Thanks for appreciation@alex-okrushko mean a lot to me,

**Comment by santoshyadav198613** [comment]
> LGTM. Once Tim approves, we'll get it merged. Thanks for working through the changes!

Thanks @brandonroberts for giving me the opportunity to work on it.
